workspaces-tests/src/bin/simple_multisig.rs

```diff
@@ -0,0 +1,126 @@
+#![allow(missing_docs)]
+
+// Ignore
+pub fn main() {}
+
+use std::fmt::Display;
+
+use near_contract_tools::{
+    approval::{
+        self,
+        simple_multisig::{AccountAuthorizer, ApprovalState, Configuration},
+        ApprovalManager,
+    },
+    rbac::Rbac,
+    slot::Slot,
+    Rbac,
+};
+use near_sdk::{
+    borsh::{self, BorshDeserialize, BorshSerialize},
+    env, near_bindgen, BorshStorageKey, PanicOnDefault,
+};
+
+#[derive(BorshSerialize, BorshStorageKey)]
+enum StorageKey {
+    SimpleMultisig,
+}
+
+#[derive(BorshSerialize, BorshDeserialize)]
+enum MyAction {
+    SayHello,
+    SayGoodbye,
+}
+
+impl approval::Action for MyAction {
+    type Output = &'static str;
+
+    fn execute(self) -> Self::Output {
+        match self {
+            Self::SayHello => "hello",
+            Self::SayGoodbye => "goodbye",
+        }
+    }
+}
+
+#[derive(BorshSerialize, BorshStorageKey)]
+enum Role {
+    Multisig,
+}
+
+#[derive(PanicOnDefault, BorshSerialize, BorshDeserialize, Rbac)]
+#[rbac(roles = "Role")]
+#[near_bindgen]
+pub struct Contract {}
+
+// This single function implementation completely implements simple multisig on
+// the contract
+impl ApprovalManager<MyAction, ApprovalState, Configuration<Self>> for Contract {
+    fn root() -> Slot<()> {
+        Slot::new(StorageKey::SimpleMultisig)
+    }
+}
+
+#[derive(Clone, Copy, Debug)]
+pub enum ApproverError {
+    UnauthorizedAccount,
+}
+
+impl Display for ApproverError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "Unauthorized account")
+    }
+}
+
+// We don't have to check env::predecessor_account_id or anything like that
```

Comment:
This is just a personal preference, but I most prefer comments that are full sentences using normal capitalization and ending with a period. I think most of your comments are that way except lack periods. The lack of period then requires extra brain load because the reader doesn't know whether the sentence is continuing on the next line because of wrapping. I'm curious whether others agree.

Reply:
@ryancwalsh I haven't thought about it explicitly until now, but now that I reflect on it I agree with you. I tend to like adding periods on the documentation, comments and there is a split second that I spend on checking if there is more to read if there is no period

Reply:
For things as pedantic as the presence of a period at the end of a comment in a test, please just fix it. I don't think it warrants a 3-person discussion and consensus.

Reply:
The discussion wouldn't be warranted for just one instance. The discussion is to ask ourselves what is the best pattern for the countless comments that we'll write in the years to come. 😃